Tools/ci-run.sh

```diff
@@ -22,10 +22,8 @@ elif [[ $OSTYPE == "linux-gnu"* ]]; then
 
   sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-$GCC_VERSION 60 $ALTERNATIVE_ARGS
 
-  export CC="gcc"
   if [[ $BACKEND == *"cpp"* ]]; then
     sudo update-alternatives --set g++ /usr/bin/g++-$GCC_VERSION
-    export CXX="g++"
   fi
 elif [[ $OSTYPE == "darwin"* ]]; then
   echo "Setting up macos compiler"
```

Comment:
Maybe remove the overrides below as well, and use CFLAGS to set the options?

Reply:
Yeah, that seems better.

Reply:
Would you like me to do that in this PR (assuming the linux variant passes CI)?

Reply:
Yes, please give it a try here.